management-api-server/src/main/java/com/datastax/mgmtapi/resources/models/EndpointStates.java

```diff
@@ -11,26 +11,11 @@
 import java.util.Map;
 import java.util.Objects;
 
-public class EndpointStates {
+public class EndpointStates extends BaseEntity {
 
   @JsonProperty(value = "entity", required = true)
   public final List<Map<String, String>> entity;
 
-  @JsonProperty(value = "variant", required = false)
-  public final Variant variant;
-
-  @JsonProperty(value = "annotations", required = false)
-  public final List<String> annotations;
-
-  @JsonProperty(value = "mediaType", required = false)
-  public final MediaType mediaType;
-
-  @JsonProperty(value = "language", required = false)
-  public final String language;
-
-  @JsonProperty(value = "encoding", required = false)
-  public final String encoding;
-
   @JsonCreator
   public EndpointStates(
       @JsonProperty("entity") List<Map<String, String>> entity,
```

Comment:
This isn't ideal. Are we sure we can't make this any more strongly typed right now with enums for the states? Will that break backwards compatibility somehow, or are we just trying to do this quickly (because we know that we're going to refactor all of it later anyway)?

Reply:
I know we discussed this earlier this morning, just want to make sure I'm understanding our motivations for taking this path, given we can retrofit schemas into these endpoints right now.

Reply:
This data is returned directly from Cassandra, so adding some enums is probably going to break at some point (unless we postprocess more from Cassandra before returning to users). We don't technically control what's returned by this endpoint, Cassandra does.

Reply:
@burmanm is right. Also, doing anything with the enums specifically will break the existing contract of the endpoint, meaning it won't be backward compatible. In a future refactoring ticket, we can introduce more strong typing and handle the case where Cassandra changes, ads or deletes an enum, which should address all the concerns here. I haven't written those tickets yet, but I will.

Reply:
I think there's a backwards compatible way to do this, but if you guys both agree to defer this work then let's keep moving forward.